testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -539,6 +540,10 @@ public void testMaterializedView()
 
         assertUpdate("DROP MATERIALIZED VIEW " + viewName);
         assertUpdate("DROP MATERIALIZED VIEW " + viewNameWithComment);
+
+        // metadata tests are called here to maintain single threaded execution, in order
+        // to avoid naming conflicts for materialized views with the same catalog name
+        testMaterializedViewsMetadata();
```

Comment:
why not just run similar queries to the one above:
```
// system.jdbc.columns with schema filter
// system.jdbc.columns with schema filter
// system.jdbc.columns with table filter
```
for the MVs already created within `testMaterializedView` test?

Reply:
It may be better to keep the MV system table tests self contained, not dependent on MVs created in `testMaterializedView` (which is quite large test already). For instance, both MVs created in `testMaterializedView` are in the same schema. `testMaterializedViewsMetadata` needs to use two MVs in seperate schemas to verify it's filtering.

We'd only call `testMaterializedViewsMetadata();` from this test method for the purpose of avoiding name collisions.

Reply:
> It may be better to keep the MV system table tests self contained, not dependent on MVs created in testMaterializedView (which is quite a large test already).

It is large, but by reusing already created MVs we wouldn't need to add a whole new bunch of test code.

> For instance, both MVs created in testMaterializedView are in the same schema. testMaterializedViewsMetadata needs to use two MVs in separate schemas to verify its filtering.

I think the same argument could be said about `information_schema` and `systems.table` filtering that already happens in `testMaterializedView` method. If we are good with filtering test there, we should also be fine with filtering for `systems.materialized_views` table.

Reply:
Sounds good, cleaned up `testMaterializedView` to maintain consistency, and added mv system table tests below.